tests/unit/test_utils.py

```diff
@@ -0,0 +1,312 @@
+        assert result == {}
+
+    def test_generate_streamlit_environment_file_with_no_requirements(self):
+        result = utils.generate_streamlit_environment_file([])
+        assert result is None
+
+    def test_generate_streamlit_environment_file(
+        self, streamlit_requirements_txt, temp_test_directory: str
+    ):
+        os.chdir(temp_test_directory)
+        result = utils.generate_streamlit_environment_file([])
+        os.chdir("..")
+
+        assert result == PosixPath("environment.yml")
+        assert os.path.isfile(os.path.join(temp_test_directory, "environment.yml"))
+
+    def test_generate_streamlit_environment_file_with_excluded_dependencies(
+        self, streamlit_requirements_txt, temp_test_directory: str
+    ):
+        os.chdir(temp_test_directory)
+        result = utils.generate_streamlit_environment_file(excluded_anaconda_deps)
+        os.chdir("..")
+        env_file = os.path.join(temp_test_directory, "environment.yml")
+        assert result == PosixPath("environment.yml")
+        assert os.path.isfile(env_file)
+        with open(env_file, "r") as f:
+            for dep in excluded_anaconda_deps:
+                assert dep not in f.read()
+
+    def test_generate_streamlit_package_wrapper(self):
+        result = utils.generate_streamlit_package_wrapper('example_stage', 'example_module', False)
+
+        assert os.path.exists(result)
+        with open(result, 'r') as f:
+            assert 'importlib.reload(sys.modules["example_module"])' in f.read()
+
+    def test_get_package_name_from_metadata_using_correct_data(
+        self, correct_metadata_file: str
+    ):
+        result = utils.get_package_name_from_metadata(correct_metadata_file)
+        assert result == "my-awesome-package"
+
+    def test_generate_snowpark_coverage_wrapper(self, temp_test_directory: str):
+        path = os.path.join(temp_test_directory, "coverage.py")
+        utils.generate_snowpark_coverage_wrapper(
+            target_file=path,
+            proc_name="process",
+            proc_signature="signature",
+            handler_module="awesomeModule",
+            handler_function="even_better_function",
+            coverage_reports_stage="example_stage",
+            coverage_reports_stage_path="nyan-cat.jpg",
+        )
+
+        assert os.path.isfile(path)
+        with open(path) as coverage_file:
+            assert (
+                "return awesomeModule.even_better_function(*args,**kwargs)"
+                in coverage_file.read()
+            )
+
+    def test_add_file_to_existing_zip(
+        self, correct_app_zip: str, correct_requirements_txt: str
+    ):
+        utils.add_file_to_existing_zip(correct_app_zip, correct_requirements_txt)
+        zip_file = ZipFile(correct_app_zip)
+
+        assert os.path.basename(correct_requirements_txt) in zip_file.namelist()
+
+    def test_install_packages(self):
+        pass  # todo: add this
+
+    def test_recursive_zip_packages(
```

Comment:
I'm wondering if this should be done here or in a followup PR but I think we need to group those tests. `utils` is very ambiguous  and we may end up with single test file with 100s of test cases for unrelated functionalities

Reply:
Probably you`re right. But my proposition is to leave it as it is - as currently we have to setup any structure for unit tests and refactor it, as the test-base grows